src/zjs_dgram.c

```diff
@@ -0,0 +1,334 @@
+
+        zjs_callback_id id = zjs_add_callback_once((jerry_value_t)user_data,
+                                      ZJS_UNDEFINED,
+                                      NULL,
+                                      NULL);
+        zjs_signal_callback(id, args, sizeof(args));
+        jerry_release_value(args[0]);
+    }
+}
+
+static jerry_value_t zjs_dgram_sock_send(const jerry_value_t function_obj,
+                                         const jerry_value_t this,
+                                         const jerry_value_t argv[],
+                                         const jerry_length_t argc)
+{
+    int ret;
+
+    if (argc < 5 || argc > 6)
+        return zjs_error("dgram.send: wrong number of args");
+
+    GET_HANDLE(dgram_handle_t, handle);
+
+    zjs_buffer_t *buf = zjs_buffer_find(argv[0]);
+    if (!buf) {
+        return zjs_error("buffer expected");
+    }
+    int offset = (int)jerry_get_number_value(argv[1]);
+    int len = (int)jerry_get_number_value(argv[2]);
+    if (offset + len > buf->bufsize) {
+        return zjs_error("offset/len beyond buffer end");
+    }
+
+    sa_family_t family = net_context_get_family(handle->udp_sock);
+    struct sockaddr sockaddr_buf;
+    get_addr(family, argv[4], argv[3], &sockaddr_buf);
+
+    struct net_buf *send_buf = net_nbuf_get_tx(handle->udp_sock);
+    if (!net_nbuf_append(send_buf, len, buf->buffer + offset)) {
+        return zjs_error("netbuf error");
+    }
+
+    void *user_data = NULL;
+    if (argc > 5) {
+        user_data = (void*)argv[5];
+    }
+
+    CHECK(net_context_sendto(send_buf, &sockaddr_buf, sizeof(sockaddr_buf),
+                             udp_sent, K_NO_WAIT, NULL, user_data));
+
+    return ZJS_UNDEFINED;
+}
+
+static jerry_value_t zjs_dgram_sock_bind(const jerry_value_t function_obj,
+                                         const jerry_value_t this,
+                                         const jerry_value_t argv[],
+                                         const jerry_length_t argc)
+{
+    int ret;
+
+    if (argc != 2)
+        return zjs_error("dgram.bind: wrong number of args");
+
+    GET_HANDLE(dgram_handle_t, handle);
+
+    sa_family_t family = net_context_get_family(handle->udp_sock);
+    struct sockaddr sockaddr_buf;
+    get_addr(family, argv[1], argv[0], &sockaddr_buf);
+
+    CHECK(net_context_bind(handle->udp_sock, &sockaddr_buf,
+                               sizeof(sockaddr_buf)));
+    // See comment in createSocket() why this is called here
+    CHECK(net_context_recv(handle->udp_sock, udp_received, K_NO_WAIT, handle));
+
+    return ZJS_UNDEFINED;
+}
+
+static jerry_value_t zjs_dgram_sock_close(const jerry_value_t function_obj,
+                                          const jerry_value_t this,
+                                          const jerry_value_t argv[],
+                                          const jerry_length_t argc)
+{
+    GET_HANDLE(dgram_handle_t, handle);
+    zjs_dgram_free_cb((uintptr_t)handle);
```

Comment:
I think we maybe need to set handle to NULL In the object so we don't double free? Or I guess while we're doing that, we set the free cb to NULL so it won't be called again when the socket goes out of scope.